core/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -320,4 +467,29 @@ public final BytesReference contentOrSourceParam() {
         }
         return BytesArray.EMPTY;
     }
+
+    /**
+     * Parses the given content type string for the media type. This method currently ignores parameters.
+     */
+    // TODO stop ignoring parameters such as charset...
+    private static XContentType parseContentType(List<String> header) {
+        if (header == null || header.isEmpty()) {
+            return null;
+        } else if (header.size() > 1) {
+            throw new IllegalArgumentException("only one Content-Type header should be provided");
+        }
+
+        String rawContentType = header.get(0);
+        final String[] elements = rawContentType.split("[ \t]*;");
+        if (elements.length > 0) {
+            final String[] splitMediaType = elements[0].split("/");
+            if (splitMediaType.length == 2 && TCHAR_PATTERN.matcher(splitMediaType[0]).matches()
+                && TCHAR_PATTERN.matcher(splitMediaType[1].trim()).matches()) {
+                return XContentType.fromMediaTypeStrict(elements[0]);
+            } else {
+                throw new IllegalArgumentException("invalid Content-Type header [" + rawContentType + "]");
+            }
+        }
+        return null;
```

Comment:
here we return null only when the split operation returns an empty array. should we rather throw exception in that case?